src/script/descriptor.cpp

```diff
@@ -356,6 +360,7 @@ class ConstPubkeyProvider final : public PubkeyProvider
     {
         return std::make_unique<ConstPubkeyProvider>(m_expr_index, m_pubkey, m_xonly);
     }
+    bool CanSelfExpand() const final { return true; }
```

Comment:
In a3be2d874503dc24b5d52fe03d4cf2c7b5ff607d _descriptor: Add CanSelfExpand()_: this can't expand. Maybe the description can clarify that the return value should be ignored if it can't expand with or without keys.

Or just just return `false`; doesn't break any tests.